### PR TITLE
Add force option to upgrade-packages gpu

### PR DESCRIPTION
**Pull request by cjohnston1158**

### CODE REVIEW

**Comment by cjohnston1158** [comment]
```
==================================================================================== 15 passed in 2748.06s (0:45:48) ====================================================================================
________________________________________________________________________________________________ summary ________________________________________________________________________________________________
  integration: commands succeeded
  congratulations :)
```

File: actions/upgrade-actions.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def _upgrade(containerd, gpu):
         raise ActionError("Must select at-least one of container and gpu")
 
     upgrade_list = _dry_run(containerd, gpu)
+    force_upgrade = False
```
Comment (addyess) [inline actions/upgrade-actions.py:80]: looks as though should be called `force_gpu_upgrade`

File: tests/integration/test_containerd_integration.py [hunk 2]
```diff
@@ -79,6 +79,38 @@ async def test_upgrade_dry_run_action(ops_test, which_action):
...
+    results = output.data.get("results", {})
+    log.info(f"Upgrade results = '{results}'")
+    assert results["containerd"]["available"] == results["containerd"]["installed"]
+    assert results["containerd"]["upgrade-available"] == "False"
+    assert not results["containerd"].get("upgrade-completed"), "No upgrade should have been run"
+    end = await process_elapsed_time(unit, "containerd")
+    assert end >= start, "containerd service shouldn't have been restarted"
+
+
+async def test_upgrade_action_without_gpu_gpu_force(ops_test):
```
Comment (addyess) [inline tests/integration/test_containerd_integration.py:98]: Would you consider a rename?

```suggestion
async def test_upgrade_action_gpu_uninstalled_but_gpu_forced(ops_test):
```

File: tests/integration/test_containerd_integration.py [hunk 3]
```diff
@@ -79,6 +79,38 @@ async def test_upgrade_dry_run_action(ops_test, which_action):
...
+    """Test running upgrade action with GPU and force without GPU does nothing."""
```
Comment (addyess) [inline tests/integration/test_containerd_integration.py:99]: Perhaps also an enlightening comment.  It'd need to be run though black.

```suggestion
    """Test running upgrade action with no GPU drivers installed.
    
        action with the `GPU` and `force` flags both set
        but without the GPU drivers installed should not upgrade any gpu drivers."""
```

File: .github/data/proxy_config.yaml [hunk 4]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (addyess) [inline .github/data/proxy_config.yaml:1]: what's this change?  Did'ja just hate that LineFeed?

No trouble here just curiosity

**Review by addyess — APPROVED** [review]
Only a small nit on a few things.  LGTM

**Comment by cjohnston1158** [comment]
```
=============================================================================================== 15 passed in 2597.77s (0:43:17) ===============================================================================================
___________________________________________________________________________________________________________ summary ___________________________________________________________________________________________________________
  integration: commands succeeded
  congratulations :)
```

File: .github/data/proxy_config.yaml (hunk 4 above)
Comment (cjohnston1158) [reply .github/data/proxy_config.yaml:1]: I guess. I had modified it locally and when I reverted I guess I removed the line.

File: actions.yaml [hunk 5]
```diff
@@ -24,3 +25,7 @@ upgrade-packages:
             description: |-
                 Upgrade gpu drivers in the event there are package updates available.
                 Note: gpu drivers are ignored if gpu_support is unecessary
+        force:
+            type: boolean
+            default: false
+            description: Only works if `gpu=true`. Force reinstall the gpu drivers.
```
Comment (addyess) [inline actions.yaml:31]: why is force only for `gpu=True`?  Why doesn't it support all the packages?

Comment (cjohnston1158) [reply actions.yaml:31]: We can, though I've never heard of a scenario where someone wanted to force containerd to reinstall.. With the situation I've gotten into though I can see a scenario where we would want to force the nvidia drivers. If you'd like I can change it though.

**Review by kwmonroe — APPROVED** [review]

**Pull request merged**